app/lib/metrics_service.rb

```diff
@@ -34,11 +34,14 @@ def self.post_to_bard(params, user)
   end
 
   def self.get_default_headers(user)
-    access_token = user.token_for_api_call
-    {
-      'Authorization' => "Bearer #{access_token.present? ? access_token.dig('access_token') : nil }",
-      'Content-Type' => 'application/json'
-    }
+    headers = { 'Content-Type' => 'application/json'}
+    if user.present? && user.registered_for_firecloud && user.token_for_api_call.present?
+      access_token = user.token_for_api_call
+      headers.merge!({
+        'Authorization' => "Bearer #{access_token.dig(:access_token)}",
+      })
+    end
+    return headers
```

Comment:
Non-blocking: Ruby has implicit returns of the last object operated on, so unless you're halting execution you don't need to say `return headers` at the end.  You can just say `headers`.